src/natives/java_util.ts

```diff
@@ -307,35 +307,131 @@ class java_util_zip_CRC32 {
 
 class java_util_zip_Deflater {
 
-  public static 'initIDs()V'(thread: JVMThread): void {
-    thread.throwNewException('Ljava/lang/UnsatisfiedLinkError;', 'Native method not implemented.');
-  }
+  public static 'initIDs()V'(thread: JVMThread): void {}
 
-  public static 'init(IIZ)J'(thread: JVMThread, arg0: number, arg1: number, arg2: number): Long {
-    thread.throwNewException('Ljava/lang/UnsatisfiedLinkError;', 'Native method not implemented.');
-    return null;
+  /**
+   * Initialize a new deflater. Using the zlib recommended default values.
+   */
+
+  public static 'init(IIZ)J'(thread: JVMThread, level: number, strategy: number, nowrap: number): Long {
+    let DEF_MEM_LEVEL = 8; // Zlib recommended default
+    let Z_DEFLATED = 8;    // This value is in the js version of pako under pako.Z_DEFLATED. 
+    // Possibly it is set to private in the Typescript version. The default value is 8, so this should work fine
+
+    let strm = new ZStreamCons();
+    let ret = deflate.deflateInit2(strm, level, Z_DEFLATED, nowrap ? -MAX_WBITS : MAX_WBITS, DEF_MEM_LEVEL, strategy);
+    
+    if (ret != ZlibReturnCode.Z_OK) {
+      let msg = ((strm.msg != null) ? strm.msg :
```

Comment:
You can just use `strm.msg` instead of `strm.msg != null` -- it'll be true whenever `strm.msg` isn't `null`, `undefined`, or `0`.